README.md

```diff
@@ -18,5 +18,153 @@ with opam. For example, to use colored terminal output you use `opam
 install ANSITerminal`. Then you must inform ocamlbuild to use the 
 ocamlfind tool : 
 `ocamlbuild -use-ocamlfind Main.byte -- tests/UnFichierDeTest.java`
-and you must add the library to the file _tags, for example : 
+and you must add the library to the file `_tags`, for example : 
 `true: package(ANSITerminal)`
+
+
+## Workflow
+
+There is three branches. 
+
+* **master** - it's the main branch. Code here should be always stable.
+* **class** - branch related to parsing of class syntax of Java.
+* **expression** - branch related to parsing of expression syntax of Java.
+
+When `class` or `expression` reaches a stable state, it should be merge into
+master branch. For that, using the 
+[pull request feature](https://help.github.com/articles/creating-a-pull-request/).
+Thanks to that, we can review your code and accept or not to integrate into master. 
+
+### Development
+
+Either you work on class or expression syntax, it should be better to create your own 
+branch based on class branch or expression branch to integrate your work. 
+
+0. Before you start to work, update your local repertory :
+
+	`git checkout class` - To be sure to be in class branch (or expression).
+
+	`git pull` - Incorporates changes from class branch of the remote repository 
+	into the current branch. Now, your class branch is up to date. 
+
+1. Create your own branch
+
+	`git checkout -b branchname` - Create for the first time your own branch 
+	and swith into it. branchname can be like that : your_name/type_of_work 
+
+	`git branch` - See your current branch.
+
+	`git checkout branchname` - Switch to the targeted branch
+ 
+2. Work hard on the project :
+
+	`git add filename` - To prepare next commit with all new contents
+	in the file.  
+
+	`git status` - Optional, but it's still better to check what you're 
+	going to commit.
+
+	`git commit` - [Write good commit message.](http://chris.beams.io/posts/git-commit/)
+	By default, you will use Vim. Try to learn basic features of 
+	[Vim](http://www.openvim.com/).
+
```

Comment:
It would be interesting to add the command 'git commit -m "message"' as an alternative to 'git commit', this way you can directory type the commit message without opening Vim or any other text editor.